src/main/java/aspects/UserRestoreAspect.java

```diff
@@ -96,16 +96,15 @@ public void closeRestoreFactory(JoinPoint joinPoint) throws Throwable {
         restoreFactory.getSQLiteDB().closeConnection();
     }
 
-    private HqAuth getAuthHeaders(String domain, String username, String sessionToken) {
-        HqAuth auth;
-        if (sessionToken != null && sessionToken.equals(authKey)) {
-            auth = new BasicAuth(touchformsUsername, touchformsPassword);
-        } else if (UserUtils.isAnonymous(domain, username)) {
+    private HqAuth getHqAuth(String domain, String username, String sessionToken) {
+        HqAuth auth = null;
+        if (UserUtils.isAnonymous(domain, username)) {
             PostgresUser postgresUser = postgresUserRepo.getUserByUsername(username);
             auth = new TokenAuth(postgresUser.getAuthToken());
-        } else {
+        } else if (sessionToken != null) {
             auth = new DjangoAuth(sessionToken);
         }
+        // Null auth expected for SMS requests
```

Comment:
we should check here that the request to formplayer used mac digest validation before we return, to prevent accidentally returning null when it shouldn't

also can we add to this comment that the meaning of a null return value here is that formplayer will use mac digest validation as auth for the requests it makes?

Reply:
That happens here https://github.com/dimagi/formplayer/blob/2fde9891f8bc8ea047c140447ffaff7d57740a8a/src/main/java/application/FormplayerAuthFilter.java#L56-L56

Reply:
what i'm trying to say in this comment and https://github.com/dimagi/formplayer/pull/661#discussion_r198451333 is that if at some later date someone introduces another type of auth that doesn't use session id and doesn't realize to update other parts of the code, it will be an issue.

i think it would be good to keep track of a single boolean named `requestValidatedWithHMAC` and make sure that any time formplayer attempts to make a request to hq that uses the mac digest validation (for both the user and the case sms workflows), it explicitly checks that it only does so when `requestValidatedWithHMAC` is true so that you can't accidentally use that with any other type of formplayer auth introduced later on

Reply:
I see what you're saying, good call. Added https://github.com/dimagi/formplayer/pull/661/commits/0391fae1e6b99d260022b26836422337a1bceba9